test/e2e/dra/dra.go

```diff
@@ -230,6 +230,39 @@ var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation]", fu
 
 				b.testPod(ctx, f.ClientSet, pod)
 			})
+
+			ginkgo.It("removes reservation from claim when pod is done", func(ctx context.Context) {
+				parameters := b.parameters()
+				pod := b.podExternal()
+				claim := b.externalClaim(allocationMode)
+				pod.Spec.Containers[0].Command = []string{"true"}
+				b.create(ctx, parameters, claim, pod)
+
+				ginkgo.By("waiting for pod to finish")
+				framework.ExpectNoError(e2epod.WaitForPodNoLongerRunningInNamespace(ctx, f.ClientSet, pod.Name, pod.Namespace), "wait for pod to finish")
+				ginkgo.By("waiting for claim to be unreserved")
+				gomega.Eventually(ctx, func(ctx context.Context) (*resourcev1alpha2.ResourceClaim, error) {
+					return f.ClientSet.ResourceV1alpha2().ResourceClaims(pod.Namespace).Get(ctx, claim.Name, metav1.GetOptions{})
+				}).WithTimeout(f.Timeouts.PodDelete).Should(gomega.HaveField("Status.ReservedFor", gomega.BeEmpty()), "reservation should have been removed")
+			})
+
+			ginkgo.It("deletes generated claims when pod is done", func(ctx context.Context) {
```

Comment:
Do we need another test that makes sure claim is not deallocated when pod is restarting after error ?

Reply:
Yes, that makes sense. The condition checking of a pod should ensure that such a pod is not treated as "done", but it's always good to verify.

I'll add it.

Reply:
[Added](https://github.com/kubernetes/kubernetes/compare/041716c3eac77c5921f575d715339378b995a9fd..e68f74fc3978920769a4791ef4fb7aca5a426e7f). As expected, nothing gets done by the resource claim controller because the pod remains active and despite being restarted.